### PR TITLE
Preventing Notices from Current Screen variables

### PR DESCRIPTION
**Pull request by bordoni**
_Resolves:_ [#30696](https://central.tri.be/issues/30696)

### CODE REVIEW

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -1594,8 +1594,12 @@ public function saved_organizers_dropdown( $current = null, $name = 'organizer[O
...
 		public function admin_body_class( $classes ) {
 			global $current_screen;
-			if ( isset( $current_screen->post_type ) &&
-				 ( $current_screen->post_type == self::POSTTYPE || $current_screen->id == 'settings_page_tribe-settings' )
+			if (
+				isset( $current_screen->post_type )
+				&& (
+					$current_screen->post_type === self::POSTTYPE
+					|| 'settings_page_tribe-settings' === $current_screen->id
+				)
```
Comment (borkweb) [inline src/Tribe/Main.php]: @jbrinley had a good idea about moving this "complex" conditional to a function in the [closed PR](https://github.com/moderntribe/the-events-calendar/pull/72#discussion_r31033115).  Here's a version of the suggestion:

```php
if ( $this->is_events_admin_screen( 'settings_page_tribe-settings' ) ) {
```

Thoughts?

Comment (bordoni) [inline src/Tribe/Main.php]: So I will do that.

Comment (borkweb) [inline src/Tribe/Main.php]: Sounds good!

File: src/Tribe/Admin/Helpers.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+
+	/**
+	 * Matcher for Admin Pages related to Post Types
+	 *
+	 * @param  boolean|string|array|null $id What will be checked to see if we return true or false
+	 *
+	 * @return boolean
+	 */
+	public function is_post_type_screen( $post_type = true ){
```
Comment (borkweb) [inline src/Tribe/Admin/Helpers.php]: I'm not sure if `$post_type = true` (or any boolean) makes sense here.  I feel like `null` would be clearer.

File: src/Tribe/Admin/Helpers.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	/**
+	 * Matcher for administration pages that are from Tribe the easier way
+	 *
+	 * @param  boolean|string|array|null $id What will be checked to see if we return true or false
+	 *
+	 * @return boolean
+	 */
+	public function is_screen( $id = true ){
```
Comment (borkweb) [inline src/Tribe/Admin/Helpers.php]: Hrm, same comment about the default being a boolean here. I feel like I'm missing your original goal with the booleans. What do they accomplish that `null` would not?

File: src/Tribe/Query.php [hunk 4]
```diff
@@ -344,8 +344,7 @@ public static function pre_get_posts( $query ) {
 			}
 
 			// if is in the admin remove the event date & upcoming filters, unless is an ajax call
-			global $current_screen;
-			if ( is_admin() && $query->tribe_is_event_query && ! empty( $current_screen->id ) && $current_screen->id == 'edit-' . Tribe__Events__Main::POSTTYPE ) {
+			if ( is_admin() && $query->tribe_is_event_query && Tribe__Events__Admin__Helpers::instance()->is_screen( 'edit-' . Tribe__Events__Main::POSTTYPE ) ) {
```
Comment (borkweb) [inline src/Tribe/Query.php]: Could this be broken down into a multi-line `if`?

Comment (bordoni) [inline src/Tribe/Query.php]: Yeaps

File: src/Tribe/Admin/Helpers.php (hunk 2 above)
Comment (bordoni) [inline src/Tribe/Admin/Helpers.php]: Ok will follow that path.

File: src/Tribe/Admin/Helpers.php [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	public function is_post_type_screen( $post_type = true ){
+		global $current_screen;
+
+		// Not in the admin we don't even care
+		if ( ! is_admin() ){
+			return false;
+		}
+
+		// Avoid Notices by checking the object type of WP_Screen
+		if ( ! is_a( $current_screen, 'WP_Screen' ) ){
```
Comment (afragen) [inline src/Tribe/Admin/Helpers.php]: Shouldn't we now be using `! $current_screen instanceof WP_Screen` instead of `is_a`?

**Comment by borkweb** [comment]
Other than the spacing funkiness that we discussed in chat, this looks good to me.

**Comment by bordoni** [comment]
I will own that funkiness and get rid of it :+1:

**Comment by bordoni** [comment]
@borkweb is this still on CR?

**Comment by borkweb** [comment]
Derp. I forgot to mark for merge. Merge away.

**Pull request merged**